frontend/src/metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper.tsx

```diff
@@ -19,24 +22,36 @@ export type LoadingAndErrorWrapperProps = {
   loadingScenes?: string[];
   renderError?: (error: any) => ReactNode;
   "data-testid"?: string;
-};
+} & (
+  | {
+      error: any;
+      loading: boolean;
+    }
+  | {
+      result: { error?: any; isLoading: boolean };
+    }
```

Comment:
this strikes me as not at all user friendly. The `loading` prop made a lot more sense than inferring loading state from some result object.